### DISCUSSION TITLE
Since November, AAB capacity has decreased and is abnormal.

### DISCUSSION THREAD

**Discussion by SEMOKISOFT** [airsdk/Adobe-Runtime-Support#2952] [Q&A] [answered]
Hello, I love Harman and Samsung very much. Thank you very much for allowing me to create apps through AS3.

One problem occurred. I am creating an AAB file through Animate and publishing the game on the Play Store.

There has been no problem at all so far, but there is a problem starting from the November version. For example, if I make my game in AAB in the September or October version before November, it originally came out to about 150 megabytes. However, if I make AAB in the version after November, it comes out to 120 megabytes. At first, I thought it was good because the capacity was reduced, but the problem is that it doesn't go up normally on the Play Console, and even if it goes up, it doesn't run when I test it in an internal test.

If you use the October or September version again, AAB will be created with 150 MB, and it will load and run well.

I've done various tests, and if you make it as an APK instead of AAB with the November version, the size comes out to 150 MB and the test play works well. Only when you make it as an AAB file does the size get drastically reduced to about 120 MB and it doesn't seem to work properly.

Therefore, we are unable to use the latest SDK and are only updating in September and October.

There is no major problem right now, but in the long run, I think I will need to use the latest SDK. Is there anyone like me?

This is a very serious problem for me. I really need help.

**Comment by SEMOKISOFT** [comment]
I am anxiously waiting for your answer. Is this problem just mine? Does anyone else have any problems? ㅠ ㅠ

**Reply by ajwfrost** [reply]
Hi - can you please check in your app descriptor file to see whether you have an entries under a `buildArchitectures` flag? I'm wondering whether it's limiting your bundle to just some of the architectures .. although that perhaps wouldn't stop it from working on a normal device (assuming the architectures match, anyway) and that also doesn't explain the APK size..

Are you able to check the logging output from ADT? If you have the AIR SDK Manager installed, go to the Troubleshooting tab and then run the publishing again from Animate, and copy/paste what you see there into here; if not, edit a file ~/.airsdk/adt.cfg to add a line "DebugOut=true" and then look for a log file generated under ~/adt.log (you can edit names etc if you need privacy or just email us the log instead, reference this issue in the subject line).

If you'd be happy sharing the AAB files from each of the builds, we could inspect these and see what the differences are.. if so, please upload using the below link:
https://transfer.harman.com/requests/N43qx0YmEB1l1Ura7kn5Bu

thanks

**Reply by SEMOKISOFT** [reply]
Oh, you're such a great person. Thank you so much for your reply. I'm not a professional programmer, so I only work within Animate. Actually, I don't understand everything you said, but I'll make sure to check and send it to you. Thank you so much because there seems to be some hope.

**Reply by ajwfrost** [reply]
:-) simplest then, if you're happy to, just send us both the working and non-working AAB files using that link, and we can try to work backwards to see what went wrong.

**Reply by SEMOKISOFT** [reply]
Based on your answer, I immediately created two AABs and sent them to you right away. Thank you so much. I respect you.

**Reply by ajwfrost** [reply]
Well I can see why the size difference, and why the new one doesn't run .. there are no native libraries included in that one at all!
That's a really strange issue .. I can't quite see how this would have happened. If you are able to get the logs for this build, it would be good - do you have the AIR SDK Manager? If not, worth installing this in order to check the troubleshooting tab and see whether there are any warnings in the build.

If you're using Animate then we won't be able to switch to just creating and Android Studio project, but we could perhaps update the configuration (via AIR SDK Manager, or the adt.cfg file?) to keep the temporary Android Studio project that we generate whilst building. If you can do that, then after publishing you should see a folder called "AndroidStudioProject" where your application/descriptor file is. Would be good to understand what's in there - perhaps if you could zip that up and send it over, we can check why there are no native libraries...?
New upload link: https://transfer.harman.com/requests/8ZBEoh9GPtNxEH7BpdOAJu

The other thing might be to try removing and re-installing the AIR SDK, as I am wondering whether something went wrong with that process for it to end up like this. Are you on Windows or macOS?

thanks

**Reply by SEMOKISOFT** [reply]
I'm using Windows 10. Your file has been sent. I suspect that the part that was originally working well has stopped working since November because the logic has changed internally.

**Reply by ajwfrost** [reply]
Hi
The file you sent was a normal - non-AIR - Android Studio project... if you look where the Animate .fla file is, do you see a folder called `AndroidStudioProject` there?
Might be best if you are able to send us that log file..
thanks

**Reply by SEMOKISOFT** [reply]
Unfortunately, no matter how hard I search that folder, I can't see the AndroidStudioProject folder. Please tell me what to do and I will do my best. Tomorrow, we will study and install the AIR SDK Manager. Please let us know and we will do our best. I am always grateful despite my busy schedule. Thank you.

**Reply by SEMOKISOFT** [reply]
[AIR 50.2.4.2](https://airsdk.harman.com/download/50.2.4.2)(19-Dec-2023)

Github-2950:

Ensuring Android ANE .so libraries are packaged in AAB bundles

I tried running it with excitement because it was included, but the capacity is still small due to the same problem.

Is this just my problem? I have no choice but to create AAB using Animate.

I wonder if there is a solution to this problem

**Reply by M4rkY4** [reply]
> I have no choice but to create AAB using Animate.

Interesting to learn why. Perhaps, using Animate for export is the main blocker?

Some time ago I've committed to a test of setting up Animate-centric project to be exported via Intellij IDEA and succeeded (by running builds on my iOS and Android devices, APM and single ANE for test were used), even created a tutorial here, but it seems that nobody has used it, many devs still prefer to stick to Animate...

Sadly, IDEA setup is not that simple, complications come from Android and some old tools used in build process, but I think you still have more chances to succeed with asking help setting up IDEA here or in Starling forum, than using Animate. It seems that the weakest point here is that we do not have here or on airsdk.dev a full tutorial on IDE set-up, perhaps it would be a nice idea for us, community, to create a tutorial on how to set-up new IDEA project.

One thing to remember is that Adobe is not interested in supporting Flash anymore, especially updating Animate for fantasies of each mobile platform, it just has a negative business value.

**Reply by FliplineStudios** [reply]
@SEMOKISOFT If Animate is the source of the problem, you could always try packaging the final AAB from the command line to see if that makes a difference.  You would take the SWF that Animate generates, and the application descriptor XML, and use the "adt -package" command from a command prompt to package up the final AAB.  This is actually how I package all of our apps, where I generate a final game SWF (from FlashDevelop in our case, but I've also done this from Animate), and then do all of the AIR packaging from the command line instead.  It might be worth a shot, at least as a workaround in the meantime until if/when Adobe updates Animate, since they seem to take forever between updates to Animate itself.

More documentation here on how to use adt on the command line:
https://airsdk.dev/docs/building/air-developer-tool/commands/package

**Reply by SEMOKISOFT** [reply]
Thank you. As you may have noticed, I am not a programmer. I used to be an animator, but I developed games and am not good at programming. So, I am making a game with animation, and I am very worried that this problem has occurred. I will study and check based on the information you provided. I will answer without going overboard. Thank you so much to everyone who posted. Happy New Year. I really like and appreciate Samsung Harman's updates, so I recently left a comment of gratitude on Samsung Lee Jae-yong's Instagram. Hahaha

**Reply by ajwfrost** [reply, marked as answer]
Hi -> so this looks like a bug in ADT, see also #2992, where we're not correctly setting the 'default' build configurations if the switch from APK to AAB was done via the override command (i.e. for Animate users... I have requested, again, to the Animate team that they give this option in the UI...)

Workaround: close Animate, find your "project-app.xml" file, and add the below line into the `<android>` section:
```
<buildArchitectures>x86,x64,armv7,armv8</buildArchitectures>
```
Then if you re-open Animate and publish for Android, it should provide all of the native libraries in the bundle.

We can get a proper fix for this out in the next release, where that `buildArchitectures` line would be implied as the default if it's not specified, which is what should have happened if Animate had actually requested an AAB via the command line...

thanks

**Reply by SEMOKISOFT** [reply]
Dear ajwfrost, thank you for your hard work and for being a great help to many programmers. Thank you so much for helping us with the solution. In Korea, 2024 is called the Year of the Blue Dragon. Always stay healthy and have good days with the energy of the Blue Dragon.
